mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/ComputedPropertyDefinition.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.entity;
+
+import com.datastax.oss.driver.internal.mapper.processor.util.generation.PropertyType;
+import com.squareup.javapoet.CodeBlock;
+import java.util.Optional;
+
+public class ComputedPropertyDefinition implements PropertyDefinition {
+
+  private final CodeBlock cqlName;
+  private final CodeBlock cqlResultName;
+  private final String getterName;
+  private final String setterName;
+  private final PropertyType type;
+
+  public ComputedPropertyDefinition(
+      String javaName,
+      Optional<String> customCqlName,
```

Comment:
It is not a good practice to pass `Optional` as a method parameter, to avoid unnecessary wrapping/unwrapping (and also because `Optional` is not serializable). I know it's already done in a few other places, but wanted to pointed that out. A better approach is to pass the string annotated with `@Nullable`.

Reply:
I'll be factoring this code out into DefaultPropertyDefinition, will update the optional parameters in that constructor when I get to it.

Reply:
> It is not a good practice to pass Optional as a method parameter

I don't 100% agree with that IDE warning. Performance is not paramount in the processor (at least not to the point where wrapping something in an optional will make a difference), and we don't store the optional in a field so serialization is not an issue (we don't serialize this object anyway for that matter).

Passing an optional here works nicely because the annotation parsing code that is used to fill that parameter already returns one.

Reply:
I agree with you, the only real bad thing is to store optionals as fields, but that is not the case here, so I'm fine if we keep them.